controllers/propagator/propagation.go

```diff
@@ -279,6 +292,147 @@ func (set decisionSet) namespaces() []string {
 	return namespaces
 }
 
+// getPolicyPlacementDecisions retrieves the placement decisions for a input
+// placement binding when the policy is bound within it.
+func (r *PolicyReconciler) getPolicyPlacementDecisions(
+	instance *policiesv1.Policy, pb policiesv1.PlacementBinding,
+) ([]appsv1.PlacementDecision, []*policiesv1.Placement, error) {
+	var decisions []appsv1.PlacementDecision
+	var placements []*policiesv1.Placement
+	var err error
+
+	subjects := pb.Subjects
+	for _, subject := range subjects {
+		if !(subject.APIGroup == policiesv1.SchemeGroupVersion.Group &&
+			subject.Kind == policiesv1.Kind &&
+			subject.Name == instance.GetName()) && !r.isPolicySetSubject(instance, subject) {
+			continue
+		}
+
+		decisions, placements, err = getPlacementDecisions(r.Client, pb, instance)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		if instance.Spec.Disabled {
+			// Only handle the first match in pb.spec.subjects
+			return nil, placements, nil
+		}
+
+		if len(decisions) == 0 {
+			log.Info("No placement decisions to process on this policy")
+		}
+
+		// Only handle the first match in pb.spec.subjects
+		break
+	}
+
+	return decisions, placements, nil
+}
+
+// containDecision checks if the input decision is present in the provided clusterDecisions,
+// If a match is found, it returns the index of the matched decision; otherwise, it returns -1.
+func containDecision(clusterDecisions []clusterDecision, decision appsv1.PlacementDecision) int {
+	for id, clusterDecision := range clusterDecisions {
+		if clusterDecision.Cluster == decision {
+			return id
+		}
+	}
+
+	return -1
+}
```

Comment:
I wonder if this kind of `clusterDecisions` thing could have been cleaner as a map of some kind. But it is not necessary to change at this point.

Reply:
yeah, agree. In our case, we don't care the order of the decisions. A map is more proper here for fast lookup and enforce uniqueness.
 As we are changing the API to` policiesv1.BindingOverrides`, we could use this map
`map[appsv1.PlacementDecision]policiesv1.BindingOverrides{}` to collect all information to be handled.

Reply:
Updated to use a map. Hope it looks cleaner now.